app/controllers/bookings_controller.rb

```diff
@@ -0,0 +1,73 @@
+class BookingsController < ApplicationController
+    before_action :find_room
+
+    def index
+        @booking = current_user.bookings
+        @sorted = @booking.order(:room_id)
+    end
+
+    def new 
+        @booking = Booking.new
+    end 
+
+    def show 
+        @bookings = Booking.find(params[:id])
+        @room = @bookings.room
+        @no_of_days = (@bookings.check_out - @bookings.check_in).to_i
+        if @no_of_days == 0
+            @no_of_days = 1
+        end
+    end
+
+    def create 
+        @booking = Booking.new(booking_param) 
+        @booking.user_id = current_user.id
+        @booking.room_id = @room.id
+        if @booking.save
+            flash.now[:success] = "booking created"
+            return redirect_to room_booking_path("#{@booking.room_id}","#{@booking.id}")
```

Comment:
and this one

Reply:
also i dont know if you need the string interpolation. can it not just be `room_booking_path(@booking.room_id, 
@booking.id)`